frontend/pulse3d/src/components/table/FilterHeader.js

```diff
@@ -0,0 +1,58 @@
+import { CheckBox } from "@mui/icons-material";
+import { useEffect, useState } from "react";
+import styled from "styled-components";
+const InputsContainer = styled.div`
+  width: 100%;
+  display: flex;
+  justify-content: space-evenly;
+`;
+export default function FilterHeader(props) {
+  const [input, setInput] = useState("");
+  const [columnName, setColumnName] = useState("");
+
+  //when collumn changes then reset all inputs
+  useEffect(() => {
+    props.setFilterColumn(columnName);
+    Array.from(document.getElementsByClassName("searchBox")).forEach((input) => {
+      if (input.id !== columnName) {
+        input.value = "";
+      }
+    });
+  }, [columnName]);
+
+  //when input changes update the filterstring
+  useEffect(() => {
+    props.setFilterString(input);
+  }, [input]);
+
+  const searchFields = props.columns.map((column, idx) => {
+    if (!props.loading) {
+      return column === "" ? (
+        <input
```

Comment:
Is there a way to just return one input component and just conditionally choose which props to pass to it? I think that would make it a little easier to read and tell what the difference between column === "" and the opposite are. There's also a lot that's the same here like onChange, type, and key, so it would be nice to not have that duplicated